ipfs/ipfs-docs
# Ci cd test/vale testing branch do not merge

@TMoMoreau feel free to make all the changes you want to this branch. 

Just don't merge this, we will delete it once we confirm that the vale GH action is actually working as expected.

Steps:
1. [install vale on your local machine](https://vale.sh/docs/vale-cli/installation/).
Confirm the installation with vale --version. Output should be something like vale version 2.20.1.
2. Clone my fork
3. Every time you are running tests on this fork, you need to rebase off of https://github.com/ElPaisano/ipfs-docs/tree/ci-cd-test/vale so that you have the latest version of the linter code i.e. in case I've made changes to that branch. This applies any changes from the vale branch to this testing branch. To do this:
    a. `git checkout ci-cd-test/vale`
    b. `git rebase ci-cd-test/vale-testing-do-not-merge`
4. Switch into this branch. You can run a `git log -3` in this branch and compare it to `git log -3` in the `ci-cd-test/vale` to confirm that the rebase worked.
5. Run `vale sync` to sync the styles and packages so that your local config matches the GH action config. You only need to do this the first time
6. Make some edits to an MD file `<file-edited>.md`
7. Run vale <file-edited>.md. If any rules defined in write-good are violated / you violated the accept and reject files, output appears.
8. For testing purposes, you can either a) fix the violations and push your changes to GH or b) leave the violations there.
9. Commit and push to this branch
10. If you chose 7a, once you push, the vale GH action will run the same rules against your PR, and will pass. If you chose 7b, the GH action will fail and report errors.

Try and make one type of change per commit i.e. if you want to test the Oxford comma rule, only make changes to the markdown that involve comma stuff.

Ping me if you have questions
